mlir/lib/CAPI/IR/BuiltinTypes.cpp

```diff
@@ -364,21 +364,22 @@ bool mlirTypeIsAVector(MlirType type) {
 MlirType mlirVectorTypeGet(intptr_t rank, const int64_t *shape,
                            MlirType elementType) {
   return wrap(VectorType::get(llvm::ArrayRef(shape, static_cast<size_t>(rank)),
-                              unwrap(elementType)));
+                              cast<ScalarTypeInterface>(unwrap(elementType))));
```

Comment:
I thought you added a builder that performs the cast, no?

Reply:
No, this PR does not have it yet. I wanted to avoid the cast in the builder so that we can have better static type checking in some places. But now the diff became to large, so maybe it's better to add it.

Reply:
Yeah I think it's not worth to be that explicit -- any type issues will be caught in the next stack frame anyway.